### DISCUSSION TITLE
Failed to generate stub file for my package

### DISCUSSION THREAD

**Discussion by zhang-qiang-github** [pybind/pybind11-stubgen#319] [Q&A / Support]
The pybind11 code to generate module is:

```
class A
{
public:
	A()
	{

	}

	/**
	* @brief explanation from c++
	*/
	int Add(int a, int b)
	{
		return a+b; 
	}

};

PYBIND11_MODULE(module, m) {


    m.doc() = "My Library Python Bindings";

	py::class_<A>(m, "A")
		.def(py::init<>())
		.def("Add", &A::Add, "explanation from pybind");

}
```

And it successfully generated a python module, and I can import it:

![image](https://github.com/user-attachments/assets/79e52934-c826-4b1e-b846-41c48ab6a553)

Then, I want to generate the stub file for this module. I have tested the `pybind11-stubgen`, and it works:
![image](https://github.com/user-attachments/assets/26efed85-c9ea-44af-9316-989a06603cd6)

However, it failed when generate stub file for my-self package: 
![image](https://github.com/user-attachments/assets/83e3beac-8cfb-42e6-bfae-502a2f49e1b2)

How should I use the `pybind11-stubgen`?

**Comment by ax3l** [comment]
I would double check you are using _the same Python interpreter_ for your `pybind11-stubgen` and your project module.

It looks to me like your `pybind11-stubgen` Python interpreter is the one in your Anaconda3 environment, while you installed your module into your system Python 3.9.20 environment.
